rasa/core/policies/rule_policy.py

```diff
@@ -224,6 +244,23 @@ def _create_loop_unhappy_lookup_from_states(
                 lookup[feature_key] = DO_NOT_PREDICT_LOOP_ACTION
         return lookup
 
+    def _check_rule_trackers(self, rule_trackers):
```

Comment:
```suggestion
    def _check_max_number_of_rule_turns(self, rule_trackers: List[TrackerWithCachedStates]) -> None:
```

Reply:
since max number of user turns in rule is only one possible restriction, I renamed it to `_check_rule_restriction`